### PR TITLE
Add annotated delly to output

### PR DESCRIPTION
**Pull request by ryanjameskennedy**
# Description
_Summary of the changes made:_
- Added `annotate_delly` module
- Added prp versions
- Added individuals mlst schemes for profiles
- Added `create_yaml` module
- Changed blastDb to pubMlstDb re mlst
- Update bonsai-prp to v0.7.1

## Primary function of PR
- [x] Minor functionality improvement

# Testing
Tested using all species and passed ci check

### CODE REVIEW

File: assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed
Comment (mhkc) [inline assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed:1]: How do this file relate to this csv file?

https://github.com/genomic-medicine-sweden/jasen/blob/master/assets/tbprofiler_dbs/converged_who_fohm_tbdb.csv

**Review by mhkc — COMMENTED** [review]
I have some questions. Lets talk about it today

File: assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed.gz
[the remark this answers is not in the document]
Comment (ryanjameskennedy) [reply assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed.gz:1]: What I've realised is that we need the latest TBProfiler singularity container ([tb-profiler:6.1.0--pyhdfd78af_1](https://depot.galaxyproject.org/singularity/tb-profiler%3A6.1.0--pyhdfd78af_1)) contains the latest fix needed to create our converged db (WHO, FoHM and tbdb). I built it by clone the TBProfiler repo and fixing it myself. Essentially, we need to pull in the latest tbprofiler image to create our db, but we use v5.0.1 to run the module.

File: assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed
Comment (ryanjameskennedy) [reply assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed:1]: The csv is used to create the `converged_who_fohm_tbdb` TBProfiler-compatible database. The `[assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed](https://github.com/genomic-medicine-sweden/jasen/pull/277/files/b95416c4beceac306e837ecbebcec04953b111c6#diff-3aef3addb2e7e1e142e1bebc1b7cb1a053076120b8b2407b3db70678b408236d)` is one of the outputs from creating the converged db. It is also bgzipped and indexed for the delly annotation.

File: assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed.gz
[the remark this answers is not in the document]
Comment (ryanjameskennedy) [reply assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed.gz:1]: So adding these files was a workaround until we use the latest TBProfiler version. I don't think it's safe to use different versions to create (v6.1.0) and run (v5.0.1) TBProfiler stuff.

File: Makefile [hunk 1]
```diff
@@ -586,6 +586,20 @@ $(MTUBE_GENOMES_DIR)/$(MTUBE_REFSEQ_ACC).fasta.bwt: $(MTUBE_GENOMES_DIR)/$(MTUBE
 # 		tb-profiler create_db --prefix converged_who_fohm_tbdb --dir $(MTUBE_TBDB_DIR)
 # 	&& 	tb-profiler load_library converged_who_fohm_tbdb --dir $(MTUBE_TBDB_DIR) |& tee -a $(INSTALL_LOG)
 
+# mtuberculosis_bgzip_bed: $(MTUBE_TBDB_DIR)/converged_who_fohm_tbdb.bed.gz
```
Comment (samuell) [inline Makefile:589]: Thought to be uncommented in an upcoming PR?

File: README.md [hunk 2]
```diff
@@ -139,10 +139,18 @@ jasentool converge --output_dir /path/to/jasen/assets/tbdb
 ### Create tbdb (ensure tb-profiler is installed)
 
 ```
+cd /path/to/jasen/assets/tbdb
```
Comment (samuell) [inline README.md:142]: This is fine, but I have started to wonder lately if we should change the `/path/to/jasen` to `$JASENPATH` or similar in various places, to make the commands more copy-and-pastable?

File: README.md [hunk 3]
```diff
@@ -139,10 +139,18 @@ jasentool converge --output_dir /path/to/jasen/assets/tbdb
...
 tb-profiler create_db --prefix converged_who_fohm_tbdb
 tb-profiler load_library converged_who_fohm_tbdb
 ```
 
+### Bgzip and index gms TBProfiler db
+
+```
+bgzip -c converged_who_fohm_tbdb.bed > /path/to/jasen/assets/tbprofiler_dbs/bed/converged_who_fohm_tbdb.bed.gz
```
Comment (samuell) [inline README.md:150]: (Save as above, re `$JASENPATH`)

File: configs/nextflow.hopper.config [hunk 4]
```diff
@@ -132,13 +138,19 @@ process {
...
 	withName: create_cdm_input {
-		container = "docker://clinicalgenomicslund/bonsai-prp:0.6.0"
+		container = "docker://clinicalgenomicslund/bonsai-prp:0.7.0"
 		publishDir = [ path: "${params.outdir}/${params.speciesDir}/cdm_input", mode: 'copy', overwrite: true ]
 	}
+	withName: create_yaml {
+		container = "${params.containerDir}/bonsai-prp.sif"
+		publishDir = [ path: "${params.outdir}/${params.speciesDir}/analysis_yaml", mode: 'copy', overwrite: true ]
+		ext.args_prp = {"${params.outdir}/${params.speciesDir}/analysis_result"}
+		ext.args_sourmash = {"${params.outdir}/${params.speciesDir}/sourmash"}
```
Comment (samuell) [inline configs/nextflow.hopper.config:152]: Not sure about this, but curious: Why are these surrounded with `{}`? From what I find online it creates closures / "anonymous functions" with the internal stuff being set. Is that the intention, and why is that?

File: configs/nextflow.ngp.config [hunk 5]
```diff
@@ -145,6 +151,12 @@ process {
 		container = "${params.containerDir}/bonsai-prp.sif"
 		publishDir = [ path: "${params.outdir}/${params.speciesDir}/cdm_input", mode: 'copy', overwrite: true ]
 	}
+	withName: create_yaml {
+		container = "${params.containerDir}/bonsai-prp.sif"
+		publishDir = [ path: "${params.outdir}/${params.speciesDir}/analysis_yaml", mode: 'copy', overwrite: true ]
+		ext.args_prp = {"${params.outdir}/${params.speciesDir}/analysis_result"}
+		ext.args_sourmash = {"${params.outdir}/${params.speciesDir}/sourmash"}
```
Comment (samuell) [inline configs/nextflow.ngp.config:158]: (Save as above re. closure syntax)

**Review by samuell — APPROVED** [review]
LGTM, with the added comments / questions.

File: Makefile (hunk 1 above)
Comment (ryanjameskennedy) [reply Makefile:589]: Next PR for sure. Needs more time...

File: README.md (hunk 2 above)
Comment (ryanjameskennedy) [reply README.md:142]: Good idea. Do we want it now or?

Comment (ryanjameskennedy) [reply README.md:142]: Feel free to push changes

Comment (samuell) [reply README.md:142]: Can do it later for sure. Better to change it everywhere in one Go in a separate PR I guess.

File: configs/nextflow.hopper.config (hunk 4 above)
Comment (samuell) [reply configs/nextflow.hopper.config:155]: Ok, well, never mind :)

File: Makefile (hunk 1 above)
Comment (samuell) [reply Makefile:589]: :+1:

File: README.md (hunk 2 above)
Comment (mhkc) [reply README.md:142]: Good idea!

**Pull request merged**